Core/Utilities/Profiler.py

```diff
@@ -0,0 +1,130 @@
+"""
+Profiling class for updated information on process status
+"""
+
+__RCSID__ = "$Id$"
+
+import psutil
+import datetime
+from DIRAC import S_OK, S_ERROR
+
+class Profiler:
+  """
+  Class for profiling both general stats about a machine and individual processes.
+  Every instance of this class is associated to a single process by using its PID.
+  Calls to the different methods of the class will return the current state of the process.
+  """
+
+  def __init__( self, pid = None ):
+    """
+    :param str pid: PID of the process to be profiled
+    """
+    self.__setProcess( pid )
+
+  def __setProcess( self, pid = None ):
+    """
+    Set the current process
+    :param str pid: PID of the process to be profiled
+    """
+    if pid:
+      self.process = psutil.Process( int( pid ) )
+    else:
+      self.process = psutil.Process()
+
+  def pid( self ):
+    """
+    Returns the process PID
+    """
+    return S_OK( self.process.pid )
+
+  def status( self ):
+    """
+    Returns the process status
+    """
+    try:
+      result = self.process.status()
+    except Exception as e:
+      return S_ERROR( e )
```

Comment:
For all the constructs like this: would you be able to specify better the exception type, and log a proper error?